docs/ref/utils.txt

```diff
@@ -514,31 +514,54 @@ Atom1Feed
     because you might be using this function outside of a view (and hence the
     current thread's locale setting will not be correct).
 
-    For cases like this, use the ``django.utils.functional.allow_lazy()``
+    For cases like this, use the ``django.utils.functional.keep_lazy()``
     decorator. It modifies the function so that *if* it's called with a lazy
     translation as one of its arguments, the function evaluation is delayed
     until it needs to be converted to a string.
 
     For example::
 
-        from django.utils.functional import allow_lazy
+        from django.utils.functional import keep_lazy, keep_lazy_text
 
         def fancy_utility_function(s, ...):
             # Do some conversion on string 's'
             ...
         # Replace unicode by str on Python 3
-        fancy_utility_function = allow_lazy(fancy_utility_function, unicode)
+        fancy_utility_function = keep_lazy(unicode)(fancy_utility_function)
+        # Or more succintly:
+        @keep_lazy(unicode)
+        def fancy_utility_function(s, ...):
+            ...
 
-    The ``allow_lazy()`` decorator takes, in addition to the function to
-    decorate, a number of extra arguments (``*args``) specifying the type(s)
-    that the original function can return. Usually, it's enough to include
-    ``unicode`` (or ``str`` on Python 3) here and ensure that your function
-    returns only Unicode strings.
+    The ``keep_lazy()`` decorator takes a number of extra arguments
+    (``*args``) specifying the type(s) that the original function can return.
+    A common use-case is to have functions that return text. For these, you can
+    just pass the ``unicode`` type to ``keep_lazy`` (or even simpler, use the
+    ``keep_lazy_text`` decorator described in the next section).
 
     Using this decorator means you can write your function and assume that the
     input is a proper string, then add support for lazy translation objects at
     the end.
 
+.. function:: keep_lazy_text(func)
+
+    A shortcut for ``keep_lazy(unicode)``.
```

Comment:
Not sure if we should refer to it as `str` (implying Python3) or `six.text_type`, `unicode` looks confusing to me.